autorest/azure/azure.go

```diff
@@ -126,6 +131,58 @@ func ExtractRequestID(resp *http.Response) string {
 	return autorest.ExtractHeaderValue(HeaderRequestID, resp)
 }
 
+// GetAsyncOperation retrieves the long-running URL to poll from the passed response.
+func GetAsyncOperation(resp *http.Response) string {
+	return resp.Header.Get(http.CanonicalHeaderKey(HeaderAsyncOperation))
+}
+
+// ResponseIsLongRunning returns true if the passed response is for an Azure long-running operation.
+func ResponseIsLongRunning(resp *http.Response) bool {
+	return autorest.ResponseRequiresPolling(resp, http.StatusCreated) && len(GetAsyncOperation(resp)) > 0
```

Comment:
@brendandixon hmm right. Although, in this case Azure APIs (specifically Storage RP) that return both `Location` and `Azure-AsyncOperation` header send different URLs in it. So there may be a behavioral difference. I'd say if the endpoint is specified as long running operation in Swagger, let's handle it in one place.

But before doing that, let's give it a try like this, I think even though these headers point to different URLs, they behaviorally do the same thing.